src/main/java/com/gzs/daos/inmemory/LanguageDaoInMemoryImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.gzs.daos.inmemory;
+
+import com.gzs.daos.LanguageDao;
+import com.gzs.data.GenerateTestData;
+import com.gzs.model.Language;
+
+import java.util.List;
+import java.util.Map;
+
+public class LanguageDaoInMemoryImpl extends InMemoryDao<Language> implements LanguageDao {
+    private static GenerateTestData generatedData;
+
+    static {
+        generatedData = GenerateTestData.getInstance();
+    }
+
+    public LanguageDaoInMemoryImpl() {
+        super();
+        this.dataMap = generatedData.getLanguages();
+    }
+
+    @Override
+    public List<Language> getAll() {
+        return getAllFromMap();
+    }
+
+    @Override
+    public Language get(int id) {
+        return getById(id);
+    }
+
+    @Override
+    public Language getByEnglishName(String englishName) {
```

Comment:
This works OK (although you could break once you find first result, if the map is large it would be wasteful to iterate further after you found the result, and also, you could get `dataMap.values()` instead of `dataMap.entrySet()`, as you don't need keys).

However, in this case it would be more elegant and readable to use stream instead of for-loop.

Same goes for `getByNativeName`

Reply:
@celicni Replaced code in all these getters (in Language,Term and Translation) with stream,like this:
`T t= dataMap.values().stream().filter(entry -> someValue.equals(entry.getSomeValue()))
.findFirst().orElse(new T());`